### ISSUE TITLE
Parent fields cause failure when deserializing child object

### ISSUE THREAD

**Issue by tanelso2** [flyx/NimYAML#131]
Hi, NimYAML gives me an error when trying to deserialize a simple object in the code below. It only seems to understand the fields that are directly defined on the type and not any parent types.

# Code:

```nim
import
    yaml

type
    Parent = object of RootObj
        i*: int
    Child = object of Parent
        s*: string

let sample = """
i: 4
s: hello
"""
var c: Child
load(sample, c)
assert c.i == 4
assert c.s == "hello"
```

# Output:

```text
  nim c -r "/home/thomas/code/yanyl/tests/tnimyaml_inheritance.nim"
Hint: used config file '/home/thomas/.choosenim/toolchains/nim-1.6.10/config/nim.cfg' [Conf]
Hint: used config file '/home/thomas/.choosenim/toolchains/nim-1.6.10/config/config.nims' [Conf]
Hint: used config file '/home/thomas/work/yanyl/tests/nim.cfg' [Conf]
Hint: gc: refc; opt: none (DEBUG BUILD, `-d:release` generates faster code)
9913 lines; 0.011s; 10.699MiB peakmem; proj: /home/thomas/work/yanyl/tests/tnimyaml_inheritance.nim; out: /home/thomas/work/yanyl/tests/tnimyaml_inheritance [SuccessX]
Hint: /home/thomas/work/yanyl/tests/tnimyaml_inheritance  [Exec]
/home/thomas/work/yanyl/tests/tnimyaml_inheritance.nim(15) tnimyaml_inheritance
/home/thomas/.nimble/pkgs/yaml-1.1.0/yaml/serialization.nim(1386) load
/home/thomas/.nimble/pkgs/yaml-1.1.0/yaml/serialization.nim(1362) construct
/home/thomas/.nimble/pkgs/yaml-1.1.0/yaml/serialization.nim(1179) constructChild
/home/thomas/.nimble/pkgs/yaml-1.1.0/yaml/serialization.nim(927) constructObject
/home/thomas/work/yanyl/tests/tnimyaml_inheritance.nim(8) constructObjectDefault
[[reraised from:
/home/thomas/work/yanyl/tests/tnimyaml_inheritance.nim(15) tnimyaml_inheritance
/home/thomas/.nimble/pkgs/yaml-1.1.0/yaml/serialization.nim(1386) load
/home/thomas/.nimble/pkgs/yaml-1.1.0/yaml/serialization.nim(1366) construct
]]
Error: unhandled exception: While constructing Child: Unknown field: "i" [YamlConstructionError]
Error: execution of an external program failed: '/home/thomas/work/yanyl/tests/tnimyaml_inheritance '
```

**Comment by flyx** [comment]
I pushed a commit that handles your test case. I am not quite confident that it works in all scenarios, and dumping is yet to be done, but you can play around with it by installing `yaml@#head`.

**Comment by flyx** [comment]
Fixed in NimYAML 2.0.0.
